File: src/charting/renderer/XAxisRendererRadarChart.java
package com.github.mikephil.charting.renderer;

import android.graphics.Canvas;

import com.github.mikephil.charting.charts.RadarChart;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.utils.MPPointF;
import com.github.mikephil.charting.utils.Utils;
import com.github.mikephil.charting.utils.ViewPortHandler;

public class XAxisRendererRadarChart extends XAxisRenderer {

    private RadarChart this.mChart;

    public XAxisRendererRadarChart(viewPortHandler: ViewPortHandler, XAxis xAxis, RadarChart chart) {
        super(viewPortHandler, xAxis, null);

        this.mChart = chart;
    }

    
    public renderAxisLabels(c: Canvas) {

        if (!mXAxis.isEnabled() || !mXAxis.isDrawLabelsEnabled())
            return;

        const labelRotationAngleDegrees = this.mXAxis.getLabelRotationAngle();
        final MPPointF drawLabelAnchor = MPPointF.getInstance(0.5f, 0.25f);

        this.mAxisLabelPaint.setTypeface(mXAxis.getTypeface());
        this.mAxisLabelPaint.setTextSize(mXAxis.getTextSize());
        this.mAxisLabelPaint.setColor(mXAxis.getTextColor());

        let sliceangle = this.mChart.getSliceAngle();

        // calculate the factor that is needed for transforming the value to
        // pixels
        let factor = this.mChart.getFactor();

        MPPointF center = this.mChart.getCenterOffsets();
        MPPointF pOut = MPPointF.getInstance(0,0);
        for (let i = 0; i < this.mChart.getData().getMaxEntryCountSet().getEntryCount(); i++) {

            let label = this.mXAxis.getValueFormatter().getAxisLabel(i, this.mXAxis);

            let angle = (sliceangle * i + this.mChart.getRotationAngle()) % 360;

            Utils.getPosition(center, this.mChart.getYRange() * factor
                    + this.mXAxis.mLabelRotatedWidth / 2f, angle, pOut);

            drawLabel(c, label, pOut.x, pOut.y - this.mXAxis.mLabelRotatedHeight / 2.f,
                    drawLabelAnchor, labelRotationAngleDegrees);
        }

        MPPointF.recycleInstance(center);
        MPPointF.recycleInstance(pOut);
        MPPointF.recycleInstance(drawLabelAnchor);
    }

	/**
	 * XAxis LimitLines on RadarChart not yet supported.
	 *
	 * @param c
	 */
	
	public renderLimitLines(c: Canvas) {
		// this space intentionally left blank
	}
}

File: src/charting/renderer/XAxisRendererRadarChart.ts
import { Canvas } from '@nativescript-community/ui-canvas';
import { RadarChart } from '../charts/RadarChart';
import { XAxis } from '../components/XAxis';
import { MPPointF } from '../utils/MPPointF';
import { Utils } from '../utils/Utils';
import { ViewPortHandler } from '../utils/ViewPortHandler';
import { XAxisRenderer } from './XAxisRenderer';

export class XAxisRendererRadarChart extends XAxisRenderer {
    private mChart: RadarChart;
    mForceLongestLabelComputation = true;

    constructor(viewPortHandler: ViewPortHandler, xAxis: XAxis, chart: RadarChart) {
        super(viewPortHandler, xAxis, null);
        this.mChart = chart;
    }

    public renderAxisLabels(c: Canvas) {
        const axis = this.xAxis;
        const chart = this.mChart;
        if (!axis.enabled || !axis.drawLabels) return;

        const labelRotationAngleDegrees = axis.labelRotationAngle;
        const drawLabelAnchor: MPPointF = { x: 0.5, y: 0.25 };
        const paint = this.axisLabelsPaint;

        paint.setFont(axis.typeface);
        paint.setTextAlign(axis.labelTextAlign);
        paint.setColor(axis.textColor);

        const sliceangle = chart.sliceAngle;

        // calculate the factor that is needed for transforming the value to
        // pixels
        const factor = chart.factor;

        const center = chart.centerOffsets;
        const pOut: MPPointF = { x: 0, y: 0 };
        const labels = axis.mLabels;
        for (let i = 0; i < chart.data.maxEntryCountSet.entryCount; i++) {
            const label = labels[i];
            if (!label) {
                continue;
            }
            const angle = (sliceangle * i + chart.rotationAngle) % 360;

            Utils.getPosition(center, chart.yRange * factor + axis.mLabelRotatedWidth / 2, angle, pOut);

            this.drawLabel(c, label, pOut.x, pOut.y - axis.mLabelRotatedHeight / 2, drawLabelAnchor, labelRotationAngleDegrees, paint);
        }

        // MPPointF.recycleInstance(center);
        // MPPointF.recycleInstance(pOut);
        // MPPointF.recycleInstance(drawLabelAnchor);
    }

    /**
     * XAxis LimitLines on RadarChart not yet supported.
     *
     * @param c
     */
    public renderLimitLines(c: Canvas) {
        // this space intentionally left blank
    }
}
